### PR TITLE
fix onBoarding sliding 

### PR DESCRIPTION
**Pull request by wafalakhal**
add sliding in the onboarding  views 
implementing Viewpager2 and TabLayout
refactoring old implementation of progress and indicators

### CODE REVIEW

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardCameraFragment.kt [hunk 1]
```diff
@@ -26,10 +26,21 @@ class OnBoardCameraFragment : BaseFragment() {
...
+        super.onResume()
+        (activity as OnBoardActivityInterface).enableSwipe(
+            isSwipeable = true,
+            isTabLayoutVisible = true
+        )
+        (activity as OnBoardActivityInterface).showButtons(
+            isNextButtonVisible = true,
+            isBackButtonVisible = true
+        )
+
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardCameraFragment.kt:40]: Extra line space :)

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardFilesFragment.kt [hunk 2]
```diff
@@ -25,24 +25,36 @@ class OnBoardFilesFragment : BaseFragment() {
...
+        (activity as OnBoardActivityInterface).showButtons(
+            isNextButtonVisible = true,
+            isBackButtonVisible = true
+        )
 
+
+    }
     override fun initView(view: View) {
-        (activity as OnBoardActivityInterface).setCurrentIndicator(2)
+      //  (activity as OnBoardActivityInterface).setCurrentIndicator(2)
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardFilesFragment.kt:42]: maybe remove the old code?

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt [hunk 3]
```diff
@@ -33,35 +42,71 @@ import rs.readahead.washington.mobile.views.dialog.uwazi.SharedLiveData.updateSe
...
-        overridePendingTransition(com.hzontal.tella_locking_ui.R.anim.`in`, com.hzontal.tella_locking_ui.R.anim.out)
+        overridePendingTransition(
+            com.hzontal.tella_locking_ui.R.anim.`in`, com.hzontal.tella_locking_ui.R.anim.out
+        )
 
         setContentView(R.layout.activity_onboarding)
 
         indicatorsContainer = findViewById(R.id.indicatorsContainer)
 
+        viewPager = findViewById(R.id.view_pager)
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt:67]: We are using `viewBinding` in the project.

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt [hunk 4]
```diff
@@ -33,35 +42,71 @@ import rs.readahead.washington.mobile.views.dialog.uwazi.SharedLiveData.updateSe
...
         initUwaziEvents()
     }
 
+    private fun initButtons() {
+
+        backBtn = findViewById(R.id.back_btn)
+        backBtn.setOnClickListener {
+            onBackPressed()
+        }
+
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt:96]: You can format the code using òption+command`enter :)

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt [hunk 5]
```diff
@@ -87,13 +132,39 @@ class OnBoardingActivity : BaseActivity(), OnBoardActivityInterface,
...
+    override fun onBackPressed() {
+
+        if (viewPager.currentItem == 0) {
+            // If the user is currently looking at the first step, allow the system to handle the
+            // Back button. This calls finish() on this activity and pops the back stack.
+            super.onBackPressed()
+        } else {
+            // Otherwise, select the previous step.
+            viewPager.currentItem = viewPager.currentItem - 1
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt:153]: make sure `viewPager.size > 0`

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt [hunk 6]
```diff
@@ -87,13 +132,39 @@ class OnBoardingActivity : BaseActivity(), OnBoardActivityInterface,
...
+
+    }
+
+
+    fun onNextPressed() {
+
+        if (viewPager.currentItem != items) {
+            // select the Next step in the viewpager
+            viewPager.currentItem = viewPager.currentItem + 1
+        }
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt:165]: what does ìtems` stands for?

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt [hunk 7]
```diff
@@ -140,22 +209,21 @@ class OnBoardingActivity : BaseActivity(), OnBoardActivityInterface,
...
-                override fun accept(option: Boolean) {
-                    if (option) {
-                        showCollectServerDialog()
-                    } else {
-                        showTellaUploadServerDialog()
-                    }
-                }
-            })*/
+            })
+        /* showDualChoiceTypeSheet(this.supportFragmentManager,
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt:213]: Same here let's remove the old code

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt [hunk 8]
```diff
@@ -249,11 +321,48 @@ class OnBoardingActivity : BaseActivity(), OnBoardActivityInterface,
...
+
+    private inner class ScreenSlidePagerAdapter(fm: FragmentManager, lifecycle: Lifecycle) :
+        FragmentStateAdapter(fm, lifecycle) {
+
+        override fun getItemCount(): Int = items
+        override fun createFragment(position: Int): Fragment {
+            var fragment: Fragment =
+                when (position) {
+                    0 -> OnBoardIntroFragment()
+                    1 -> OnBoardCameraFragment()
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt:357]: Let's use an enum instead :)

**Review by djordjemarusic — APPROVED** [review]

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt [hunk 9]
```diff
@@ -29,89 +34,135 @@ import rs.readahead.washington.mobile.views.dialog.CollectServerDialogFragment.C
...
-    private val isOnboardLockSet by lazy { intent.getBooleanExtra(IS_ONBOARD_LOCK_SET, false)  }
+    private var viewpagerItemsCount = 0
+    private val isFromSettings by lazy { intent.getBooleanExtra(IS_FROM_SETTINGS, false) }
+    private val isOnboardLockSet by lazy { intent.getBooleanExtra(IS_ONBOARD_LOCK_SET, false) }
     private val presenter by lazy { OnBoardPresenter(this) }
-    private lateinit var indicatorsContainer: LinearLayout
+    private lateinit var backBtn: TextView
+    private lateinit var nextBtn: TextView
+    private lateinit var binding: ActivityOnboardingBinding
+    val ONBOARDING_INTRODUCTION_VIEW_INDEX = 0
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt:49]: This should be constants not an attributes in the class.

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt [hunk 10]
```diff
@@ -29,89 +34,135 @@ import rs.readahead.washington.mobile.views.dialog.CollectServerDialogFragment.C
...
+                OnBoardLockSetFragment(), R.id.rootOnboard
             )
         }
         initUwaziEvents()
     }
 
-    private fun setupIndicators(indicatorCount : Int) {
-        indicatorsContainer.removeAllViews()
+    private fun initButtons() {
+        backBtn = findViewById(R.id.back_btn)
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt:82]: where is the viewBinding here :)

**Review by ahlem-jarrar — CHANGES_REQUESTED** [review]

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt (hunk 10 above)
Comment (ahlem-jarrar) [reply mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt:82]: @wafalakhal also resolve this comments if it's fixed.

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardCameraFragment.kt [hunk 11]
```diff
@@ -13,36 +13,34 @@ class OnBoardCameraFragment : BaseFragment() {
     private lateinit var backBtn: TextView
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardCameraFragment.kt:13]: why initializing this ? while already using viewBinding ?

Comment (ahlem-jarrar) [reply mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardCameraFragment.kt:13]: @wafalakhal please don't forget to resolve the comments.

File: mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt [hunk 12]
```diff
@@ -29,89 +34,135 @@ import rs.readahead.washington.mobile.views.dialog.CollectServerDialogFragment.C
...
     private val presenter by lazy { OnBoardPresenter(this) }
-    private lateinit var indicatorsContainer: LinearLayout
+    private lateinit var backBtn: TextView
+    private lateinit var nextBtn: TextView
+    private lateinit var binding: ActivityOnboardingBinding
+    val ONBOARDING_INTRODUCTION_VIEW_INDEX = 0
+    val ONBOARDING_CAMERA_VIEW_INDEX = 1
+    val ONBOARDING_RECORDER_VIEW_INDEX = 2
+    val ONBOARDING_FILES_VIEW_INDEX = 3
+    val ONBOARDING_LOCK_VIEW_INDEX = 4
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/activity/onboarding/OnBoardingActivity.kt:53]: this should be private as you only need it in this class.

**Review by ahlem-jarrar — CHANGES_REQUESTED** [review]

**Review by ahlem-jarrar — APPROVED** [review]

**Pull request merged**